compiler/rustc_hir/src/definitions.rs

```diff
@@ -417,11 +421,17 @@ impl Definitions {
     }
 
     #[inline(always)]
-    pub fn local_def_path_hash_to_def_id(&self, hash: DefPathHash) -> Option<LocalDefId> {
+    pub fn local_def_path_hash_to_def_id(&self, hash: DefPathHash) -> LocalDefId {
+        debug_assert!(hash.stable_crate_id() == self.stable_crate_id);
```

Comment:
I don't think this assert is necessary - if the crate ID doesn't match, then the map lookup should panic.

Reply:
Yes, the lookup would panic but it would be harder to determine the cause of that. If you insist I'll take the assertion out -- but I think it might help catch errors early.